src/cuda/Framework/EventBatch.h

```diff
@@ -0,0 +1,36 @@
+#ifndef EventBatch_h
+#define EventBatch_h
+
+#include <vector>
+
+#include "Framework/Event.h"
+#include "Framework/EventRange.h"
+
+namespace edm {
+
+  class EventBatch {
+  public:
+    Event& emplace(int streamId, int eventId, ProductRegistry const& reg) {
+      events_.emplace_back(streamId, eventId, reg);
+      return events_.back();
+    }
+
+    void reserve(size_t capacity) { events_.reserve(capacity); }
+
+    void clear() { events_.clear(); }
+
+    size_t size() const { return events_.size(); }
+
+    bool empty() const { return events_.empty(); }
+
+    EventRange range() { return EventRange(&events_.front(), &events_.back() + 1); }
+
+    ConstEventRange range() const { return ConstEventRange(&events_.front(), &events_.back() + 1); }
```

Comment:
Not very relevant for prototyping itself, but i think being able to use something along
```suggestion
    EventRange range() { return EventRange(events_.begin(), events_.end()); }

    ConstEventRange range() const { return ConstEventRange(events_.begin(), events_.end()); }
```
would be clearer to read.

Reply:
The reason I didn't do that was to avoid making `EventRange` and `ConstEventRange` use `std::vector<Event>::iterator` and `const_iterator`: then it could only work for a range of events stored in a `vector`, while using plain pointers it can work with any consecutive range of events.

An alternative could be to add a constructor like
```c++
template <typename IT>
EventRange(IT begin, IT end) : begin_(std::addressof(*begin)), end_(std::addressof((*end)) { ... }
```
but then one could be tempted to pass non-contiguous iterators.
Maybe one could add something like
```c++
  assert((end - begin) == (end_ - begin_));
```
or,  once we move to C++20 with concepts, we could use `contiguous_iterator` ?

Reply:
For the purposes of the prototype the current pointers are fine for me. With C++20 maybe even passing the range as `std::span` could be an option.